### ISSUE TITLE
[package] snap package example fails to parse plugs

### ISSUE THREAD

**Issue by crotwell** [jreleaser/jreleaser#1686]
### Task List

- [X] Steps to reproduce provided
- [X] Stacktrace (if present) provided
- [X] Example that reproduces the problem (link to git repository is ideal)
- [X] Full description of the issue provided (see below)

### Steps to Reproduce

1. Copy packager yaml from example docs for snap to jreleaser.yml
2. fix indentation for plugs
3. run jreleaser config

### Expected Behaviour

config should parse successfully and print. Similar issue with slots. Removing both the plugs and slots sections
seems to allow the config to parse ok.

### Actual Behaviour

```
$ jreleaser config
jreleaser 1.12.0. Consider becoming a sponsor at https://opencollective.com/jreleaser
[INFO]  JReleaser 1.12.0
[INFO]  Configuring with jreleaser.yml
[INFO]    - basedir set to /Users/crotwell/tmp
[INFO]    - outputdir set to /Users/crotwell/tmp/out/jreleaser
[INFO]  Reading configuration
Unexpected error parsing config file. jreleaser.yml
Cannot deserialize value of type `java.util.ArrayList<org.jreleaser.model.internal.packagers.SnapPackager$Plug>` from Object value (token `JsonToken.START_OBJECT`)
 at [Source: (BufferedInputStream); line: 207, column: 7] (through reference chain: org.jreleaser.model.internal.JReleaserModel["packagers"]->org.jreleaser.model.internal.packagers.Packagers["snap"]->org.jreleaser.model.internal.packagers.SnapPackager["plugs"])
```

### Environment Information

- **Operating System**: OSX 14.5
- **JReleaser Version:** 1.12.0
- **JDK Version:** 21.0.3 (Homebrew 21.0.3)

Please paste stacktraces from `[out|build|target]/jreleaser/trace.log`.

```
[INFO]  JReleaser 1.12.0
------------------------------------------------------------
jreleaser 1.12.0
------------------------------------------------------------
Build time:   2024-05-01 12:32:47Z
Revision:     8f5795140149238b358a64e43fa5b8f83e55f64d
JVM:          21.0.3 (Homebrew 21.0.3)
------------------------------------------------------------
[INFO]  Configuring with jreleaser.yml
[INFO]    - basedir set to /Users/crotwell/tmp
[INFO]    - outputdir set to /Users/crotwell/tmp/out/jreleaser
[INFO]  Reading configuration
org.jreleaser.model.JReleaserException: Unexpected error parsing config file. jreleaser.yml
        at org.jreleaser.config.JReleaserConfigLoader.loadConfig(JReleaserConfigLoader.java:52)
        at org.jreleaser.engine.context.ContextCreator.resolveModel(ContextCreator.java:100)
        at org.jreleaser.cli.AbstractModelCommand.createContext(AbstractModelCommand.java:145)
        at org.jreleaser.cli.AbstractModelCommand.execute(AbstractModelCommand.java:89)
        at org.jreleaser.cli.AbstractCommand.call(AbstractCommand.java:52)
        at org.jreleaser.cli.AbstractModelCommand.call(AbstractModelCommand.java:53)
        at org.jreleaser.cli.AbstractCommand.call(AbstractCommand.java:37)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
        at picocli.CommandLine.execute(CommandLine.java:2170)
        at org.jreleaser.cli.Main.execute(Main.java:98)
        at org.jreleaser.cli.Main.run(Main.java:81)
        at org.jreleaser.cli.Main.main(Main.java:70)
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.util.ArrayList<org.jreleaser.model.internal.packagers.SnapPackager$Plug>` from Object value (token `JsonToken.START_OBJECT`)
 at [Source: (BufferedInputStream); line: 207, column: 7] (through reference chain: org.jreleaser.model.internal.JReleaserModel["packagers"]->org.jreleaser.model.internal.packagers.Packagers["snap"]->org.jreleaser.model.internal.packagers.SnapPackager["plugs"])
        at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59)
        at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1767)
        at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1541)
        at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1488)
        at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.handleNonArray(CollectionDeserializer.java:402)
        at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:254)
        at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:30)
        at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:310)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
        at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:310)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
        at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:310)
        at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
        at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342)
        at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4905)
        at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3885)
        at org.jreleaser.config.yaml.YamlJReleaserConfigParser.parse(YamlJReleaserConfigParser.java:86)
        at org.jreleaser.config.JReleaserConfigLoader.loadConfig(JReleaserConfigLoader.java:50)
        ... 17 more
```

**Comment by aalmiray** [comment]
Is this a problem with the documentation or the actual template?

https://jreleaser.org/guide/latest/reference/packagers/snap.html

**Comment by crotwell** [comment]
Not sure, the docs for sure, but implementation also seems to have problems. 

After some playing around this seemed to parse correctly, but seems kind of verbose so maybe I am missing something with how yaml works. Note also similar issue with `slots`.

```
    plugs:
      - name: some_plug_name
        attributes:
          key: value
        reads:
          - $HOME/.foo
        writes:
          - $HOME/.foo
          - $HOME/.foobar
```

But then the plugs section in the output snapcraft.yaml is missing the `reads` and `writes` sections:

```
plugs:
  some_plug_name:
    key: value
```

**Comment by crotwell** [comment]
Possibly related, but `grade` and `base` do not seems to be updating from config. I have tried base of `core`,  `core18` and `core22` but it always ends up as `core20` and grade of `devel` becomes `stable`:
```
packagers:
  snap:
    active: ALWAYS
    remoteBuild: true
    grade: devel
    base: core
    plugs:
      - name: some_plug_name
        attributes:
          key: value
        reads:
          - $HOME/.foo
        writes:
          - $HOME/.foo
          - $HOME/.foobar
```

but the output shows:
```
grade: stable
confinement: strict
base: core20
type: app
```

Also, snap now has core22 available and so maybe worth updating docs list of "Supported values"?

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.14.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.14.0))
